c10/core/StorageImpl.h

```diff
@@ -192,22 +173,14 @@ struct C10_API StorageImpl final : public c10::intrusive_ptr_target {
   void UniqueStorageShareExternalPointer(
       at::DataPtr&& data_ptr,
       const caffe2::TypeMeta& data_type,
-      size_t capacity) {
-    data_type_ = data_type;
-    // TODO: Use CAFFE_ENFORCE_WITH_CALLER equivalent
-    // For now causes lots of redefine issues if caffe2/core/logging.h is used
-    if (data_type_.id() == caffe2::TypeIdentifier::uninitialized()) {
-      AT_ERROR(
-          "To share with a raw external pointer you need to have meta "
-          "already set.");
-    }
-    data_ptr_ = std::move(data_ptr);
-    // NOTE: data_type might change and so it's also possible that capacity
-    // might not be divisible by itemsize. There is no way for us to keep track
-    // of the exact capacity if we're not explicity storing is. More conrectely
-    // capacity() might not return the value that was set here, if itemsize does
-    // not evenly divide it.
-    numel_ = capacity / data_type_.itemsize();
+      size_t capacity);
+
+  void set_received_cuda(bool received_cuda) {
```

Comment:
Is it possible to push the setting of this to the constructor? If I understand correctly, once a Storage is constructed, its `received_cuda_` property can never change. Having a setter here just hanging out makes that invariant less obvious.

Reply:
Not easy, and I can guess for the same reason resizable(false) is set outside of constructor too.